rapids-cmake/export/detail/component.cmake

```diff
@@ -0,0 +1,80 @@
+#=============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_export_component
+-----------------------
+
+.. versionadded:: v23.04.00
+
+.. code-block:: cmake
+
+Generate the necessary -Config.cmake modules needed for an optional component
+of a rapids export package.
+
+  rapids_export_component( (BUILD|INSTALL) <project_name> <component_name> <export_set> <unique_name> <namespace>)
+
+The :cmake:command:`rapids_export_component` function generates
+the `<proj>-<unique_name>-targets.cmake` and `<proj>-<unique_name>-dependencies.cmake`
+files necessary for :cmake:command:`rapids_export` to support optional
+components.
+
+.. note::
+  It is an anti-pattern to use this function directly. The vast majority of projects
+  would use :cmake:command:`rapids_export`
+
+
+#]=======================================================================]
+# cmake-lint: disable=W0105,R0913
+function(rapids_export_component type project_name component_name export_set unique_name namespace)
```

Comment:
For me to learn, when do you prefer `cmake_parse_arguments` to just writing out the function signature? AFAICT you use the simpler approach when you only have a bunch of single value arguments with no multi value or options?

Reply:
Where possible I prefer to avoid `cmake_parse_arguments` for `detail/` methods so that we can keep the complexity of those down.

In general I try to only use `cmake_parse_arguments` when we have optional arguments or multiple arguments that take multi-values. This way we get nicer error messages when calling functions without the required arguments.